### PR TITLE
Design for adding a replacement referee

### PR DESCRIPTION
**Pull request by fofr**
Design outlined below.

What's missing:
- Updates to the emails we send out

Journeys:
```
Email asking for new referee > Click link > Sign in > Click sign in link > See ask for new details page
```

```
Homepage > Sign in > Click sign in link > See ask for new details page
```

### Ask for new details
- Show this page instead of the default home page
- Give reason why they need to replace referee, matching words used in emails
- Allow users to skip the step

localhost_3000_applications_new-referee

### Give new details
- Same as referee form with minor changes
- Removes hint text about when we will contact them
- Has a descriptive title that matches button used to reach here (and no caption)

localhost_3000_application_12345_references_alternative

### Confirm referee details before submitting
- Because the action is immediate, we will automatically email new referee, we need a review step
- Give the candidate a chance to catch any errors
- Confirming will send email and return user to application

localhost_3000_application_12345_references_alternative-review

### If they skip past the first screen
If they skipped the first screen, or didn't add a referee, maintain a warning on the dashboard

localhost_3000_applications (1)

### CODE REVIEW

File: app/views/applications/new-referee.njk [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{% extends "_content.njk" %}
+{% set applicationPath = "/application/12345" %}
+
+{% block primary %}
+  <h1 class="govuk-heading-xl">You need to give details of a new referee</h1>
```
Comment (adamsilver) [inline app/views/applications/new-referee.njk]: ‘Add a new referee’ (and make consistent in email)

File: app/views/applications/new-referee.njk [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  {# <p>Our email requesting a reference didn’t reach Glinda Baumbach.</p> #}
+  {# <p>We haven’t had a reference from Doyle Denesik.</p> #}
+  <p>Carmel Kohler said they won’t give a reference.</p>
+
+  <p>We can’t send your application to your teacher training providers without 2 complete references.</p>
+
+  <div class="govuk-!-margin-top-6 govuk-!-margin-bottom-4">
+    {{ govukButton({
+      text: "Give details of a new referee",
```
Comment (adamsilver) [inline app/views/applications/new-referee.njk]: ‘Continue’ or ‘Add a new referee’ or ‘Continue to add a new referee’

File: app/views/applications/new-referee.njk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      href: applicationPath + "/references/alternative"
+    }) }}
+  </div>
+
+  <p>
+    <a href="/applications">Continue without adding a referee</a>
```
Comment (adamsilver) [inline app/views/applications/new-referee.njk]: ‘Continue without adding a new referee’

File: app/views/application/references/alternative-review.njk [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    classes: "govuk-!-margin-bottom-6",
+    headingLevel: 3,
+    titleText: "New referee",
+    html: refereeHtml
+  }) }}
+
+  <p class="govuk-body">We’ll contact your new referee immediately.</p>
+
+  {{ govukButton({
+    text: "Confirm new referee",
```
Comment (adamsilver) [inline app/views/application/references/alternative-review.njk]: Reads a little oddly to me.

How about ‘Submit new referee’

File: app/views/application/references/alternative-review.njk [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  {{ govukBackLink({
+    href: applicationPath + "/references/alternative",
+    text: "Back"
+  }) }}
+{% endblock %}
+
+{% block primary %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-xl">Check your answers before&nbsp;confirming</h1>
```
Comment (adamsilver) [inline app/views/application/references/alternative-review.njk]: ‘Check the referee's details before submitting’

File: app/views/application/references/alternative.njk [hunk 6]
```diff
@@ -0,0 +1,64 @@
+{% extends "_form.njk" %}
+
+{% set type = applicationValue(["referees", id, "type"]) | lower %}
+{% set title = "Give details for your new referee" %}
```
Comment (adamsilver) [inline app/views/application/references/alternative.njk]: ‘Details of new referee’ or ‘Add details of new referee’

**Comment by fofr** [comment]
> what happens if the user continues without adding a referee?

They see the application dashboard, with a warning message (see last screenshot)

> What do they see if they login again?

They see the same message again

File: app/views/application/references/alternative.njk (hunk 6 above)
Comment (fofr) [reply app/views/application/references/alternative.njk]: Going for "Add a new referee"

File: app/views/application/references/alternative-review.njk (hunk 5 above)
Comment (fofr) [reply app/views/application/references/alternative-review.njk]: Going for "Check your referee’s details before confirming"

File: app/views/application/references/alternative-review.njk (hunk 4 above)
Comment (fofr) [reply app/views/application/references/alternative-review.njk]: Neither sit well for me. Thought I can't think of an alternative.
I've also thought about mentioning that contact is immediate with something like "Confirm and contact" or "Confirm and send".

**Pull request merged**